### PR TITLE
[MOB-6229] Reposition close x button

### PR DESCRIPTION
**Pull request by pauljung14**
## JIRA Ticket(s) if any

* [MOB-6229](https://iterable.atlassian.net/browse/MOB-6229)

## Description

Close button was being positioned relative to document rather than iframe boundaries.
For non-Safari browsers, return close x button to the iframe document.
For Safari browsers, keep close button on the main document and dynamically reposition relative to iframe boundaries.

## Test Steps

In both Safari and non-Safari browsers:
1. Send in-app to self
2. Call getMessages in application
3. Expected: close x button appears over message iframe according to closeButton params
4. Expected: resizing window keeps close x button over message iframe

[MOB-6229]: https://iterable.atlassian.net/browse/MOB-6229?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/inapp/inapp.ts [hunk 1]
```diff
@@ -93,128 +90,23 @@ export function getInAppMessages(
...
-      };
-    } catch (err: any) {
-      console.warn(
-        'Error requesting in-app messages',
-        err?.response?.data?.clientErrors ?? err
-      );
-    }
-    return await requestInAppMessages({});
-  };
-
```
Comment (pauljung14) [inline src/inapp/inapp.ts:207]: These have been moved to ./request.ts

File: src/inapp/inapp.ts [hunk 2]
```diff
@@ -405,48 +289,29 @@ export function getInAppMessages(
...
-                position: fixed;
-                top: 0;
-                left: 0;
-                z-index: -1;
-              `;
-            /* 
-              don't let the user tab to this button. 
-              It's not necessarily for blind folks to tab over 
-            */
-            absoluteDismissButton.tabIndex = -1;
```
Comment (pauljung14) [inline src/inapp/inapp.ts:434]: This is handled now by `generateAbsoluteDismissButton` in ./utils.ts

File: src/index.ts [hunk 3]
```diff
@@ -1,7 +1,6 @@
 export * from 'src/authorization';
 export * from './users';
 export * from './inapp';
-export * from './request';
```
Comment (pauljung14) [inline src/index.ts:4]: Had to remove from index due to circular dependency

File: src/constants.ts [hunk 4]
```diff
@@ -40,6 +40,10 @@ export const STATIC_HEADERS = {
 /* how long animations fade/side in for. */
 export const ANIMATION_DURATION = 400;
 
+export const DEFAULT_CLOSE_BUTTON_OFFSET_PERCENTAGE = 4;
+export const CLOSE_X_BUTTON_ID = 'close-x-button';
+export const ABSOLUTE_DISMISS_BUTTON_ID = 'absolute-dismiss-button';
+
```
Comment (mprew97) [inline src/constants.ts:46]: 🙌

File: src/inapp/tests/inapp.test.ts [hunk 5]
```diff
@@ -325,12 +325,12 @@ describe('getInAppMessages', () => {
 
       expect(frame?.tagName).toBe('IFRAME');
 
-      const element = document.body.querySelector(
-        '[data-qa-custom-close-button]'
-      );
+      const closeButton = (
```
Comment (mprew97) [inline src/inapp/tests/inapp.test.ts:328]: good rename 👍

**Review by mprew97 — APPROVED** [review]
Tested and lgtm!

**Pull request merged**